File: Olix/Arrays/day-3/removeElement.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function (nums, val) {
  let k = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] === val) continue;
    nums[k++] = nums[i];
  }
  return k;
};

console.log(removeElement([1, 1, 1, 1, 2, 2, 3], 1));

File: Mark/Lists/Day-3/remove element/remove-element.py

class Solution:
    def removeElement(self, nums: list[int], val: int) -> int:
        for num in list(nums):
            if num == val:
                nums.remove(num)
            else:
                continue
        
        print(nums)


solution = Solution()
output = solution.removeElement([0,1,2,2,3,0,4,2,2,2,2], 2)
print(output)
